### DISCUSSION TITLE
How to set timeout for individual requests?

### DISCUSSION THREAD

**Discussion by artworkad** [RicoSuter/NSwag#4315] [Q&A]
How can the timeout be set per request for the c# client generated by nSwag?

To apply a timeout for all requests, I do:

```
builder.Services
    .AddHttpClient("MyApi", client =>
    {
        client.BaseAddress = new Uri(builder.HostEnvironment.BaseAddress);
        client.Timeout = TimeSpan.FromMinutes(5);
    })
    .AddHttpMessageHandler<BaseAddressAuthorizationMessageHandler>();
```

**Comment by gmkado** [comment]
@artworkad did you figure out a way?  I'm trying to accomplish the same thing

**Reply by artworkad** [reply]
No, seems like it's not possible.

**Comment by rhegner** [comment]
I did it this way. In a partial class I implemented

```
private Task<HttpClient> CreateHttpClientAsync(CancellationToken cancellationToken)
    => Task.FromResult(HttpClientFactory.CreateClient());
```

This way I get a fresh `HttpClient` for every request. Make sure `"useHttpClientCreationMethod": true` is set in your config.

Then I can adjust the timeout for specific requests:

```
private async Task PrepareRequestAsync(HttpClient client, HttpRequestMessage request, string url, CancellationToken cancellationToken)
{
    if (request.Method.Method == "POST" && url.StartsWith("..."))
    {
        client.Timeout = TimeSpan.FromHours(1);
    }
}
```

**Comment by arshrp** [comment]
The NSwag generated methods all get an overload with a CancellationToken parameter...so you could send a different token with a different timeout on each call.  Apologies, as I probably misunderstand the question.
